default.nix

```diff
@@ -1,22 +1,33 @@
 let
-  config = {
-    packageOverrides = super: rec {
+  thisOverlay = self: super: {
+    my-R = self.rWrapper.override {
+      packages = with self.rPackages; [ dplyr ];
+    };
 
-      haskellPackages = super.haskellPackages.override {
-        overrides = haskellPackagesNew: haskellPackagesOld: {
-          nix-mwe = haskellPackagesNew.callPackage ./cabal.nix {
-            R = super.rWrapper.override {
-              packages = with super.rPackages; [ dplyr ];
-            };
-          };
-        };
-      };
+    haskellPackages = super.haskellPackages.override {
+      overrides = hself: hsuper: {
+        nix-mwe = self.haskell.lib.overrideCabal (hself.callCabal2nix "nix-mwe" ./. {}) (oldAttrs: {
```

Comment:
Here I use `callCabal2nix`.  This is much easier than having a separate `cabal.nix` file.